### PR TITLE
Parallel in jenkins

### PR DESCRIPTION
**Pull request by jgedeon-osi**
Making build go faster

### CODE REVIEW

**Comment by aaronOSI** [comment]
must go faster, MUST GO FASTER

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -19,9 +19,11 @@
 #
 # See http://rubydoc.info/gems/rspec-core/RSpec/Core/Configuration
 require 'webmock/rspec'
+require 'rspec_junit_formatter'
```
Comment (thomasramirez) [inline spec/spec_helper.rb]: move to .rspec?

File: docker/test/docker-compose.yml [hunk 2]
```diff
@@ -53,5 +53,3 @@ services:
       - redis
   redis:
     image: 'redis:3.0'
-    ports:
```
Comment (thomasramirez) [inline docker/test/docker-compose.yml]: :thumbsup:

File: Makefile [hunk 3]
```diff
@@ -21,17 +21,20 @@ test:
...
-	@ docker-compose $(TEST_ARGS) build --pull rspec_test
-	@ docker-compose $(TEST_ARGS) build lint javascript_test
+	@ docker-compose $(TEST_ARGS) build --pull rspec_test &
+	@ docker-compose $(TEST_ARGS) build lint &
+	@ docker-compose $(TEST_ARGS) build javascript_test &
+	@ wait
+	${INFO} "Running lint..."
+	@ docker-compose $(TEST_ARGS) up lint
+	@ docker cp $$(docker-compose $(TEST_ARGS) ps -q lint):/reports/. reports
+	@ $(call check_exit_code,$(TEST_ARGS),lint)
```
Comment (thomasramirez) [inline Makefile]: Not a huge fan of stopping tests after one test type fails but, given that all tests should be run before CI runs and the CI lint is the same as local lint, I suppose it's okay.

File: Makefile [hunk 4]
```diff
@@ -21,17 +21,20 @@ test:
...
 	${INFO} "Running tests..."
 	@ docker-compose $(TEST_ARGS) up rspec_test
 	@ docker cp $$(docker-compose $(TEST_ARGS) ps -q rspec_test):/reports/. reports
-	@ docker-compose $(TEST_ARGS) up lint
-	@ docker cp $$(docker-compose $(TEST_ARGS) ps -q lint):/reports/. reports
 	@ docker-compose $(TEST_ARGS) up javascript_test
 	@ docker cp $$(docker-compose $(TEST_ARGS) ps -q javascript_test):/reports/. reports
 	@ $(call check_exit_code,$(TEST_ARGS),rspec_test)
```
Comment (thomasramirez) [inline Makefile]: Given that rspec occasionally behaves differently in CI than local, I'd prefer that the JS tests run even if rspec fails.

File: spec/spec_helper.rb (hunk 1 above)
Comment (jgedeon-osi) [inline spec/spec_helper.rb]: Done

File: Makefile [hunk 5]
```diff
@@ -47,10 +50,11 @@ build:
...
 release:
 	${INFO} "Pulling latest images..."
-	@ docker-compose $(RELEASE_ARGS) pull
+	@ docker-compose $(RELEASE_ARGS) pull &
 	${INFO} "Building images..."
-	@ docker-compose $(RELEASE_ARGS) build app
-	@ docker-compose $(RELEASE_ARGS) build --pull nginx
+	@ docker-compose $(RELEASE_ARGS) build app &
+	@ docker-compose $(RELEASE_ARGS) build --pull nginx &
+	@ wait
```
Comment (brunoOSI) [inline Makefile]: not a fan of the time managing here, is this getting a status code when done or something?

Comment (brunoOSI) [inline Makefile]: @jgedeon-osi clarified that waits for everything so it's ok

**Pull request merged**